File: Python/6KYU/GA-DE-RY-PO-LU-KI2.py
def encode(str, key): return ende(str, key)
def decode(str, key): return ende(str, key)

def ende(str, key):
    key+=key.upper()
    rkey = ""
    for l in range(0,len(key),2):
        rkey+=key[l+1]+key[l]
    str = list(str)
    for i in range(0,len(str)):
        if str[i] in key:
            str[i] = rkey[list(key).index(str[i])]
    return "".join(str)

File: Ruby/6KYU/GA-DE-RY-PO-LU-KI2.rb
def encode(str, key) ende(str, key) end
def decode(str, key) ende(str, key) end

def ende(str, key)
  key+=key.upcase
  rkey = ""
  (0..key.length-1).step(2).each {|l| rkey+=key[l+1]+key[l]}
  str.tr(key,rkey)
end